draft-ietf-cose-bls-key-representations.md

```diff
@@ -215,36 +219,145 @@ An example JWK for the Bls12381 curve where the public key is in the G1 subgroup
 }
 ```
 
-Another example of a different JWK for the Bls12381 curve where the public key is in the G1 subgroup.
+An example JWK for the Bls12381 curve where the public key is in the G2 subgroup.
 
 ```
 {
   "kty": "OKP",
-  "crv": "Bls12381G1",
-  "d": "PV21Ysd3RNtDBzx94WOkIItSdMkMx0xdjtVFWen9xy8",
-  "x": "jQb7AerHCU1Zf7oUCMYioqAkK_Q35-hDmg9cKhIJzGyZmQgb4saO376vjmGkvaLC"
+  "crv": "Bls12381G2",
+  "d": "CUrC9Xp5pEonbFaykalWlbNYYwueJlcuoOexhEhpu0k",
+  "x": "rvdKcdkxwlj0Y-XZsFpz1hDPJGjnLN27IJipbmaLlaKdYfICGG6dzakG6EkdcvW0AtVV6hXBSKtdFnKQKmmD759tMYYuvKYf5o2cZnROLN5iWQ2H6vp6FlLi71a_AE5I"
 }
 ```
 
-An example JWK for the Bls12381 curve where the public key is in the G2 subgroup.
+### Bls48581 Key Pairs
+
+The following examples showcase JWKs for both the G1 and G2 subgroups of the Bls48581 curve. As before, note that the examples also include the corresponding private key, expressed through the inclusion of the “d” parameter.
+
+An example JWK for the Bls48581 curve where the public key is in the G1 subgroup.
 
 ```
 {
   "kty": "OKP",
-  "crv": "Bls12381G2",
-  "d": "CUrC9Xp5pEonbFaykalWlbNYYwueJlcuoOexhEhpu0k",
-  "x": "rvdKcdkxwlj0Y-XZsFpz1hDPJGjnLN27IJipbmaLlaKdYfICGG6dzakG6EkdcvW0AtVV6hXBSKtdFnKQKmmD759tMYYuvKYf5o2cZnROLN5iWQ2H6vp6FlLi71a_AE5I"
+  "crv": "Bls48581G1",
+  "x": "AwwXxiUemAF5ZYORdFabm87Y-8mzSW8jB1HZ-0BvXi87VHlTJbUs7yFEZlYNxcaP2uUnfgBhR7GUgKK_uiW_yQ2v5EeNuitTKcE",
+  "d": "IbTrWWx6KkM1ggh-53OMB8Q8Bv3jXRJ3mGf0-v4YcrrGCf6joOTKRUujsccKUqHckiITeqWmGmdGgKJ3luPmFTh0i95oV2Cktw"
```

Comment:
Why are the x and d values of different lengths?  That seems suspicious to me.

If it's omitting leading zeros, for instance, that's not what JWKs do.

Reply:
The length of the private key is given by the field size while the compressed representation of the public key (using the x parameter) is derived from the group size, I'll add some text to clarify this.